.generator/src/generator/templates/model/ObjectSerializer.j2

```diff
@@ -127,24 +127,42 @@ export class ObjectSerializer {
       const attributesMap = typeMap[type].getAttributeTypeMap();
       const instance: {[index: string]: any} = {};
 
-      for (const attributeName in attributesMap) {
+      for (const attributeName in data) {
         const attributeObj = attributesMap[attributeName];
-        if (attributeName == "additionalProperties") {
-          if (data.additionalProperties) {
-            for (const key in data.additionalProperties) {
-              instance[key] = ObjectSerializer.serialize(
-                data.additionalProperties[key],
-                attributeObj.type,
-                attributeObj.format
-              );
-            }
-          }
+        if (attributeName === "_unparsed" || attributeName === "additionalProperties") {
           continue;
+        } else if (attributeObj) {
+          instance[attributeObj.baseName] = ObjectSerializer.serialize(
+            data[attributeName],
+            attributeObj.type,
+            attributeObj.format
+          );
+        } else {
+          throw new Error('unknown attribute ' + attributeName + ' of type ' + type);
         }
```

Comment:
the error is what actually fixes the oneOf selection, since the raised error tells the try/catch block in the previous function to skip the oneOf variant instead of incrementing the number of "successful" serializations. Do you think it would be better to add a flag on the serialize function or maybe separate serialize function specifically for oneOfs?